### PR TITLE
Add Slack node

### PR DESCRIPTION
**Pull request by chewcw**
Add Slack node for sending message to a specific channel in a workspace. Input action should looks like:
```json
{
  "topic": "SEND.V1",
  "payload": {
      "message": "message",
      "channel": "channel",
      "workspace": "workspace"
  }
}
```

### CODE REVIEW

**Comment by chewcw** [comment]
Removed channel name XPath selector and expects an input like this:
```json
{
  "topic": "SEND.V1",
  "payload": {
      "message": "message",
      "channel": "https://xxxx.slack.com/archives/C8529AP33"
  }
}
```

File: nodes/slack/slack.common.ts [hunk 1]
```diff
@@ -6,8 +6,7 @@ const Schema = {
   send: Message.extend({
     topic: z.literal('SEND.V1'),
     payload: z.object({
-      workspace: z.string().nonempty(),
-      channel: z.string().nonempty(),
+      channelLink: z.string().nonempty(),
```
Comment (danielepolencic) [inline nodes/slack/slack.common.ts:9]: this should be:

```ts
channelLink: z.string().transform(it => {
  try {
    const url = new URL(it)
    return url.toString()
  } catch {
    return ''
  }
}).refine(it => it.length > 0)
```

File: nodes/slack/slack.lib.ts [hunk 2]
```diff
@@ -13,57 +13,69 @@ export function Setup(setupArg: SetupArg) {
...
           node.status({
             fill: 'green',
             shape: 'dot',
-            text: `Last processed ${workspace}: ${action.payload.channel} ${time()}`,
+            text: `Last processed send message to ${channelLink} ${time()}`,
           })
         } catch (e) {
           browser?.disconnect()
-          const message = [`Error writing message to channel ${channel}`, JSON.stringify(e)].join('\n')
+          const message = [`Error writing message to channelLink ${channelLink}`, e].join('\n')
```
Comment (danielepolencic) [inline nodes/slack/slack.lib.ts:78]: `e` is printed as `[object Object]`

File: nodes/slack/slack.lib.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          send(Event.failed({ message }))
+
+          node.error(message)
+          node.status({ fill: 'red', shape: 'dot', text: `Error ${time()}` })
+        }
+
+        return done()
+      }
+      default:
+        break
```
Comment (danielepolencic) [inline nodes/slack/slack.lib.ts:89]: I think here we need to call `done()`. If we don't do that, any message with a non `SEND.V1` topic will block the node forever.

**Review by danielepolencic — APPROVED** [review]

**Pull request merged**